internal/dinosql/config.go

```diff
@@ -58,43 +59,78 @@ type Override struct {
 	GoType string `json:"go_type"`
 
 	// fully qualified name of the Go type, e.g. `github.com/segmentio/ksuid.KSUID`
-	PostgresType string `json:"postgres_type"`
+	DBType                  string `json:"db_type"`
+	Deprecated_PostgresType string `json:"postgres_type"`
+
+	// for global overrides only when two different engines are in use
+	Engine Engine `json:"engine,omitempty"`
 
 	// True if the GoType should override if the maching postgres type is nullable
 	Null bool `json:"null"`
 
 	// fully qualified name of the column, e.g. `accounts.id`
 	Column string `json:"column"`
 
-	columnName  string
-	table       pg.FQN
-	goTypeName  string
+	ColumnName  string
+	Table       pg.FQN
+	GoTypeName  string
 	goPackage   string
 	goBasicType bool
 }
 
+func (c *GenerateSettings) GatherEngines() map[Engine]struct{} {
+	engines := map[Engine]struct{}{}
+	for _, pkg := range c.Packages {
+		if _, ok := engines[pkg.Engine]; !ok {
+			engines[pkg.Engine] = struct{}{}
+		}
+	}
+	return engines
+}
+
+func (c *GenerateSettings) ValidateGlobalOverrides() error {
+	engines := c.GatherEngines()
+	usesMultipleEngines := len(engines) > 1
+	for _, oride := range c.Overrides {
+		if usesMultipleEngines && oride.Engine == "" {
+			return fmt.Errorf(`the "engine" field is required for global type overrides because your configuration uses multiple database engines`)
+		}
+	}
+	return nil
+}
+
 func (o *Override) Parse() error {
+
+	// validate deprecated postgres_type field
+	if o.Deprecated_PostgresType != "" {
+		color.Yellow(`WARNING: "postgres_type" is deprecated. Instead, use "db_type" to specify a type override.`)
```

Comment:
These errors messages are great. Explicit and helpful.

I appreciate the thought of adding color output, but I'd like to do that in a single, holistic pass in the future. Can you just write this to stderr without color?